<!DOCTYPE html>
    <html>
    <head>
        <meta charset="UTF-8">
        <title>BIO334 Theory</title>
        <style>
/* From extension vscode.github */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

.vscode-dark img[src$=\#gh-light-mode-only],
.vscode-light img[src$=\#gh-dark-mode-only] {
	display: none;
}

/* From extension vscode.markdown-math */
@font-face{font-family:KaTeX_AMS;font-style:normal;font-weight:400;src:url(fonts/KaTeX_AMS-Regular.woff2) format("woff2"),url(fonts/KaTeX_AMS-Regular.woff) format("woff"),url(fonts/KaTeX_AMS-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Caligraphic;font-style:normal;font-weight:700;src:url(fonts/KaTeX_Caligraphic-Bold.woff2) format("woff2"),url(fonts/KaTeX_Caligraphic-Bold.woff) format("woff"),url(fonts/KaTeX_Caligraphic-Bold.ttf) format("truetype")}@font-face{font-family:KaTeX_Caligraphic;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Caligraphic-Regular.woff2) format("woff2"),url(fonts/KaTeX_Caligraphic-Regular.woff) format("woff"),url(fonts/KaTeX_Caligraphic-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Fraktur;font-style:normal;font-weight:700;src:url(fonts/KaTeX_Fraktur-Bold.woff2) format("woff2"),url(fonts/KaTeX_Fraktur-Bold.woff) format("woff"),url(fonts/KaTeX_Fraktur-Bold.ttf) format("truetype")}@font-face{font-family:KaTeX_Fraktur;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Fraktur-Regular.woff2) format("woff2"),url(fonts/KaTeX_Fraktur-Regular.woff) format("woff"),url(fonts/KaTeX_Fraktur-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Main;font-style:normal;font-weight:700;src:url(fonts/KaTeX_Main-Bold.woff2) format("woff2"),url(fonts/KaTeX_Main-Bold.woff) format("woff"),url(fonts/KaTeX_Main-Bold.ttf) format("truetype")}@font-face{font-family:KaTeX_Main;font-style:italic;font-weight:700;src:url(fonts/KaTeX_Main-BoldItalic.woff2) format("woff2"),url(fonts/KaTeX_Main-BoldItalic.woff) format("woff"),url(fonts/KaTeX_Main-BoldItalic.ttf) format("truetype")}@font-face{font-family:KaTeX_Main;font-style:italic;font-weight:400;src:url(fonts/KaTeX_Main-Italic.woff2) format("woff2"),url(fonts/KaTeX_Main-Italic.woff) format("woff"),url(fonts/KaTeX_Main-Italic.ttf) format("truetype")}@font-face{font-family:KaTeX_Main;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Main-Regular.woff2) format("woff2"),url(fonts/KaTeX_Main-Regular.woff) format("woff"),url(fonts/KaTeX_Main-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Math;font-style:italic;font-weight:700;src:url(fonts/KaTeX_Math-BoldItalic.woff2) format("woff2"),url(fonts/KaTeX_Math-BoldItalic.woff) format("woff"),url(fonts/KaTeX_Math-BoldItalic.ttf) format("truetype")}@font-face{font-family:KaTeX_Math;font-style:italic;font-weight:400;src:url(fonts/KaTeX_Math-Italic.woff2) format("woff2"),url(fonts/KaTeX_Math-Italic.woff) format("woff"),url(fonts/KaTeX_Math-Italic.ttf) format("truetype")}@font-face{font-family:"KaTeX_SansSerif";font-style:normal;font-weight:700;src:url(fonts/KaTeX_SansSerif-Bold.woff2) format("woff2"),url(fonts/KaTeX_SansSerif-Bold.woff) format("woff"),url(fonts/KaTeX_SansSerif-Bold.ttf) format("truetype")}@font-face{font-family:"KaTeX_SansSerif";font-style:italic;font-weight:400;src:url(fonts/KaTeX_SansSerif-Italic.woff2) format("woff2"),url(fonts/KaTeX_SansSerif-Italic.woff) format("woff"),url(fonts/KaTeX_SansSerif-Italic.ttf) format("truetype")}@font-face{font-family:"KaTeX_SansSerif";font-style:normal;font-weight:400;src:url(fonts/KaTeX_SansSerif-Regular.woff2) format("woff2"),url(fonts/KaTeX_SansSerif-Regular.woff) format("woff"),url(fonts/KaTeX_SansSerif-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Script;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Script-Regular.woff2) format("woff2"),url(fonts/KaTeX_Script-Regular.woff) format("woff"),url(fonts/KaTeX_Script-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Size1;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Size1-Regular.woff2) format("woff2"),url(fonts/KaTeX_Size1-Regular.woff) format("woff"),url(fonts/KaTeX_Size1-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Size2;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Size2-Regular.woff2) format("woff2"),url(fonts/KaTeX_Size2-Regular.woff) format("woff"),url(fonts/KaTeX_Size2-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Size3;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Size3-Regular.woff2) format("woff2"),url(fonts/KaTeX_Size3-Regular.woff) format("woff"),url(fonts/KaTeX_Size3-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Size4;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Size4-Regular.woff2) format("woff2"),url(fonts/KaTeX_Size4-Regular.woff) format("woff"),url(fonts/KaTeX_Size4-Regular.ttf) format("truetype")}@font-face{font-family:KaTeX_Typewriter;font-style:normal;font-weight:400;src:url(fonts/KaTeX_Typewriter-Regular.woff2) format("woff2"),url(fonts/KaTeX_Typewriter-Regular.woff) format("woff"),url(fonts/KaTeX_Typewriter-Regular.ttf) format("truetype")}.katex{text-rendering:auto;font:normal 1.21em KaTeX_Main,Times New Roman,serif;line-height:1.2;text-indent:0}.katex *{-ms-high-contrast-adjust:none!important;border-color:currentColor}.katex .katex-version:after{content:"0.13.24"}.katex .katex-mathml{clip:rect(1px,1px,1px,1px);border:0;height:1px;overflow:hidden;padding:0;position:absolute;width:1px}.katex .katex-html>.newline{display:block}.katex .base{position:relative;white-space:nowrap;width:-webkit-min-content;width:-moz-min-content;width:min-content}.katex .base,.katex .strut{display:inline-block}.katex .textbf{font-weight:700}.katex .textit{font-style:italic}.katex .textrm{font-family:KaTeX_Main}.katex .textsf{font-family:KaTeX_SansSerif}.katex .texttt{font-family:KaTeX_Typewriter}.katex .mathnormal{font-family:KaTeX_Math;font-style:italic}.katex .mathit{font-family:KaTeX_Main;font-style:italic}.katex .mathrm{font-style:normal}.katex .mathbf{font-family:KaTeX_Main;font-weight:700}.katex .boldsymbol{font-family:KaTeX_Math;font-style:italic;font-weight:700}.katex .amsrm,.katex .mathbb,.katex .textbb{font-family:KaTeX_AMS}.katex .mathcal{font-family:KaTeX_Caligraphic}.katex .mathfrak,.katex .textfrak{font-family:KaTeX_Fraktur}.katex .mathtt{font-family:KaTeX_Typewriter}.katex .mathscr,.katex .textscr{font-family:KaTeX_Script}.katex .mathsf,.katex .textsf{font-family:KaTeX_SansSerif}.katex .mathboldsf,.katex .textboldsf{font-family:KaTeX_SansSerif;font-weight:700}.katex .mathitsf,.katex .textitsf{font-family:KaTeX_SansSerif;font-style:italic}.katex .mainrm{font-family:KaTeX_Main;font-style:normal}.katex .vlist-t{border-collapse:collapse;display:inline-table;table-layout:fixed}.katex .vlist-r{display:table-row}.katex .vlist{display:table-cell;position:relative;vertical-align:bottom}.katex .vlist>span{display:block;height:0;position:relative}.katex .vlist>span>span{display:inline-block}.katex .vlist>span>.pstrut{overflow:hidden;width:0}.katex .vlist-t2{margin-right:-2px}.katex .vlist-s{display:table-cell;font-size:1px;min-width:2px;vertical-align:bottom;width:2px}.katex .vbox{align-items:baseline;display:inline-flex;flex-direction:column}.katex .hbox{width:100%}.katex .hbox,.katex .thinbox{display:inline-flex;flex-direction:row}.katex .thinbox{max-width:0;width:0}.katex .msupsub{text-align:left}.katex .mfrac>span>span{text-align:center}.katex .mfrac .frac-line{border-bottom-style:solid;display:inline-block;width:100%}.katex .hdashline,.katex .hline,.katex .mfrac .frac-line,.katex .overline .overline-line,.katex .rule,.katex .underline .underline-line{min-height:1px}.katex .mspace{display:inline-block}.katex .clap,.katex .llap,.katex .rlap{position:relative;width:0}.katex .clap>.inner,.katex .llap>.inner,.katex .rlap>.inner{position:absolute}.katex .clap>.fix,.katex .llap>.fix,.katex .rlap>.fix{display:inline-block}.katex .llap>.inner{right:0}.katex .clap>.inner,.katex .rlap>.inner{left:0}.katex .clap>.inner>span{margin-left:-50%;margin-right:50%}.katex .rule{border:0 solid;display:inline-block;position:relative}.katex .hline,.katex .overline .overline-line,.katex .underline .underline-line{border-bottom-style:solid;display:inline-block;width:100%}.katex .hdashline{border-bottom-style:dashed;display:inline-block;width:100%}.katex .sqrt>.root{margin-left:.27777778em;margin-right:-.55555556em}.katex .fontsize-ensurer.reset-size1.size1,.katex .sizing.reset-size1.size1{font-size:1em}.katex .fontsize-ensurer.reset-size1.size2,.katex .sizing.reset-size1.size2{font-size:1.2em}.katex .fontsize-ensurer.reset-size1.size3,.katex .sizing.reset-size1.size3{font-size:1.4em}.katex .fontsize-ensurer.reset-size1.size4,.katex .sizing.reset-size1.size4{font-size:1.6em}.katex .fontsize-ensurer.reset-size1.size5,.katex .sizing.reset-size1.size5{font-size:1.8em}.katex .fontsize-ensurer.reset-size1.size6,.katex .sizing.reset-size1.size6{font-size:2em}.katex .fontsize-ensurer.reset-size1.size7,.katex .sizing.reset-size1.size7{font-size:2.4em}.katex .fontsize-ensurer.reset-size1.size8,.katex .sizing.reset-size1.size8{font-size:2.88em}.katex .fontsize-ensurer.reset-size1.size9,.katex .sizing.reset-size1.size9{font-size:3.456em}.katex .fontsize-ensurer.reset-size1.size10,.katex .sizing.reset-size1.size10{font-size:4.148em}.katex .fontsize-ensurer.reset-size1.size11,.katex .sizing.reset-size1.size11{font-size:4.976em}.katex .fontsize-ensurer.reset-size2.size1,.katex .sizing.reset-size2.size1{font-size:.83333333em}.katex .fontsize-ensurer.reset-size2.size2,.katex .sizing.reset-size2.size2{font-size:1em}.katex .fontsize-ensurer.reset-size2.size3,.katex .sizing.reset-size2.size3{font-size:1.16666667em}.katex .fontsize-ensurer.reset-size2.size4,.katex .sizing.reset-size2.size4{font-size:1.33333333em}.katex .fontsize-ensurer.reset-size2.size5,.katex .sizing.reset-size2.size5{font-size:1.5em}.katex .fontsize-ensurer.reset-size2.size6,.katex .sizing.reset-size2.size6{font-size:1.66666667em}.katex .fontsize-ensurer.reset-size2.size7,.katex .sizing.reset-size2.size7{font-size:2em}.katex .fontsize-ensurer.reset-size2.size8,.katex .sizing.reset-size2.size8{font-size:2.4em}.katex .fontsize-ensurer.reset-size2.size9,.katex .sizing.reset-size2.size9{font-size:2.88em}.katex .fontsize-ensurer.reset-size2.size10,.katex .sizing.reset-size2.size10{font-size:3.45666667em}.katex .fontsize-ensurer.reset-size2.size11,.katex .sizing.reset-size2.size11{font-size:4.14666667em}.katex .fontsize-ensurer.reset-size3.size1,.katex .sizing.reset-size3.size1{font-size:.71428571em}.katex .fontsize-ensurer.reset-size3.size2,.katex .sizing.reset-size3.size2{font-size:.85714286em}.katex .fontsize-ensurer.reset-size3.size3,.katex .sizing.reset-size3.size3{font-size:1em}.katex .fontsize-ensurer.reset-size3.size4,.katex .sizing.reset-size3.size4{font-size:1.14285714em}.katex .fontsize-ensurer.reset-size3.size5,.katex .sizing.reset-size3.size5{font-size:1.28571429em}.katex .fontsize-ensurer.reset-size3.size6,.katex .sizing.reset-size3.size6{font-size:1.42857143em}.katex .fontsize-ensurer.reset-size3.size7,.katex .sizing.reset-size3.size7{font-size:1.71428571em}.katex .fontsize-ensurer.reset-size3.size8,.katex .sizing.reset-size3.size8{font-size:2.05714286em}.katex .fontsize-ensurer.reset-size3.size9,.katex .sizing.reset-size3.size9{font-size:2.46857143em}.katex .fontsize-ensurer.reset-size3.size10,.katex .sizing.reset-size3.size10{font-size:2.96285714em}.katex .fontsize-ensurer.reset-size3.size11,.katex .sizing.reset-size3.size11{font-size:3.55428571em}.katex .fontsize-ensurer.reset-size4.size1,.katex .sizing.reset-size4.size1{font-size:.625em}.katex .fontsize-ensurer.reset-size4.size2,.katex .sizing.reset-size4.size2{font-size:.75em}.katex .fontsize-ensurer.reset-size4.size3,.katex .sizing.reset-size4.size3{font-size:.875em}.katex .fontsize-ensurer.reset-size4.size4,.katex .sizing.reset-size4.size4{font-size:1em}.katex .fontsize-ensurer.reset-size4.size5,.katex .sizing.reset-size4.size5{font-size:1.125em}.katex .fontsize-ensurer.reset-size4.size6,.katex .sizing.reset-size4.size6{font-size:1.25em}.katex .fontsize-ensurer.reset-size4.size7,.katex .sizing.reset-size4.size7{font-size:1.5em}.katex .fontsize-ensurer.reset-size4.size8,.katex .sizing.reset-size4.size8{font-size:1.8em}.katex .fontsize-ensurer.reset-size4.size9,.katex .sizing.reset-size4.size9{font-size:2.16em}.katex .fontsize-ensurer.reset-size4.size10,.katex .sizing.reset-size4.size10{font-size:2.5925em}.katex .fontsize-ensurer.reset-size4.size11,.katex .sizing.reset-size4.size11{font-size:3.11em}.katex .fontsize-ensurer.reset-size5.size1,.katex .sizing.reset-size5.size1{font-size:.55555556em}.katex .fontsize-ensurer.reset-size5.size2,.katex .sizing.reset-size5.size2{font-size:.66666667em}.katex .fontsize-ensurer.reset-size5.size3,.katex .sizing.reset-size5.size3{font-size:.77777778em}.katex .fontsize-ensurer.reset-size5.size4,.katex .sizing.reset-size5.size4{font-size:.88888889em}.katex .fontsize-ensurer.reset-size5.size5,.katex .sizing.reset-size5.size5{font-size:1em}.katex .fontsize-ensurer.reset-size5.size6,.katex .sizing.reset-size5.size6{font-size:1.11111111em}.katex .fontsize-ensurer.reset-size5.size7,.katex .sizing.reset-size5.size7{font-size:1.33333333em}.katex .fontsize-ensurer.reset-size5.size8,.katex .sizing.reset-size5.size8{font-size:1.6em}.katex .fontsize-ensurer.reset-size5.size9,.katex .sizing.reset-size5.size9{font-size:1.92em}.katex .fontsize-ensurer.reset-size5.size10,.katex .sizing.reset-size5.size10{font-size:2.30444444em}.katex .fontsize-ensurer.reset-size5.size11,.katex .sizing.reset-size5.size11{font-size:2.76444444em}.katex .fontsize-ensurer.reset-size6.size1,.katex .sizing.reset-size6.size1{font-size:.5em}.katex .fontsize-ensurer.reset-size6.size2,.katex .sizing.reset-size6.size2{font-size:.6em}.katex .fontsize-ensurer.reset-size6.size3,.katex .sizing.reset-size6.size3{font-size:.7em}.katex .fontsize-ensurer.reset-size6.size4,.katex .sizing.reset-size6.size4{font-size:.8em}.katex .fontsize-ensurer.reset-size6.size5,.katex .sizing.reset-size6.size5{font-size:.9em}.katex .fontsize-ensurer.reset-size6.size6,.katex .sizing.reset-size6.size6{font-size:1em}.katex .fontsize-ensurer.reset-size6.size7,.katex .sizing.reset-size6.size7{font-size:1.2em}.katex .fontsize-ensurer.reset-size6.size8,.katex .sizing.reset-size6.size8{font-size:1.44em}.katex .fontsize-ensurer.reset-size6.size9,.katex .sizing.reset-size6.size9{font-size:1.728em}.katex .fontsize-ensurer.reset-size6.size10,.katex .sizing.reset-size6.size10{font-size:2.074em}.katex .fontsize-ensurer.reset-size6.size11,.katex .sizing.reset-size6.size11{font-size:2.488em}.katex .fontsize-ensurer.reset-size7.size1,.katex .sizing.reset-size7.size1{font-size:.41666667em}.katex .fontsize-ensurer.reset-size7.size2,.katex .sizing.reset-size7.size2{font-size:.5em}.katex .fontsize-ensurer.reset-size7.size3,.katex .sizing.reset-size7.size3{font-size:.58333333em}.katex .fontsize-ensurer.reset-size7.size4,.katex .sizing.reset-size7.size4{font-size:.66666667em}.katex .fontsize-ensurer.reset-size7.size5,.katex .sizing.reset-size7.size5{font-size:.75em}.katex .fontsize-ensurer.reset-size7.size6,.katex .sizing.reset-size7.size6{font-size:.83333333em}.katex .fontsize-ensurer.reset-size7.size7,.katex .sizing.reset-size7.size7{font-size:1em}.katex .fontsize-ensurer.reset-size7.size8,.katex .sizing.reset-size7.size8{font-size:1.2em}.katex .fontsize-ensurer.reset-size7.size9,.katex .sizing.reset-size7.size9{font-size:1.44em}.katex .fontsize-ensurer.reset-size7.size10,.katex .sizing.reset-size7.size10{font-size:1.72833333em}.katex .fontsize-ensurer.reset-size7.size11,.katex .sizing.reset-size7.size11{font-size:2.07333333em}.katex .fontsize-ensurer.reset-size8.size1,.katex .sizing.reset-size8.size1{font-size:.34722222em}.katex .fontsize-ensurer.reset-size8.size2,.katex .sizing.reset-size8.size2{font-size:.41666667em}.katex .fontsize-ensurer.reset-size8.size3,.katex .sizing.reset-size8.size3{font-size:.48611111em}.katex .fontsize-ensurer.reset-size8.size4,.katex .sizing.reset-size8.size4{font-size:.55555556em}.katex .fontsize-ensurer.reset-size8.size5,.katex .sizing.reset-size8.size5{font-size:.625em}.katex .fontsize-ensurer.reset-size8.size6,.katex .sizing.reset-size8.size6{font-size:.69444444em}.katex .fontsize-ensurer.reset-size8.size7,.katex .sizing.reset-size8.size7{font-size:.83333333em}.katex .fontsize-ensurer.reset-size8.size8,.katex .sizing.reset-size8.size8{font-size:1em}.katex .fontsize-ensurer.reset-size8.size9,.katex .sizing.reset-size8.size9{font-size:1.2em}.katex .fontsize-ensurer.reset-size8.size10,.katex .sizing.reset-size8.size10{font-size:1.44027778em}.katex .fontsize-ensurer.reset-size8.size11,.katex .sizing.reset-size8.size11{font-size:1.72777778em}.katex .fontsize-ensurer.reset-size9.size1,.katex .sizing.reset-size9.size1{font-size:.28935185em}.katex .fontsize-ensurer.reset-size9.size2,.katex .sizing.reset-size9.size2{font-size:.34722222em}.katex .fontsize-ensurer.reset-size9.size3,.katex .sizing.reset-size9.size3{font-size:.40509259em}.katex .fontsize-ensurer.reset-size9.size4,.katex .sizing.reset-size9.size4{font-size:.46296296em}.katex .fontsize-ensurer.reset-size9.size5,.katex .sizing.reset-size9.size5{font-size:.52083333em}.katex .fontsize-ensurer.reset-size9.size6,.katex .sizing.reset-size9.size6{font-size:.5787037em}.katex .fontsize-ensurer.reset-size9.size7,.katex .sizing.reset-size9.size7{font-size:.69444444em}.katex .fontsize-ensurer.reset-size9.size8,.katex .sizing.reset-size9.size8{font-size:.83333333em}.katex .fontsize-ensurer.reset-size9.size9,.katex .sizing.reset-size9.size9{font-size:1em}.katex .fontsize-ensurer.reset-size9.size10,.katex .sizing.reset-size9.size10{font-size:1.20023148em}.katex .fontsize-ensurer.reset-size9.size11,.katex .sizing.reset-size9.size11{font-size:1.43981481em}.katex .fontsize-ensurer.reset-size10.size1,.katex .sizing.reset-size10.size1{font-size:.24108004em}.katex .fontsize-ensurer.reset-size10.size2,.katex .sizing.reset-size10.size2{font-size:.28929605em}.katex .fontsize-ensurer.reset-size10.size3,.katex .sizing.reset-size10.size3{font-size:.33751205em}.katex .fontsize-ensurer.reset-size10.size4,.katex .sizing.reset-size10.size4{font-size:.38572806em}.katex .fontsize-ensurer.reset-size10.size5,.katex .sizing.reset-size10.size5{font-size:.43394407em}.katex .fontsize-ensurer.reset-size10.size6,.katex .sizing.reset-size10.size6{font-size:.48216008em}.katex .fontsize-ensurer.reset-size10.size7,.katex .sizing.reset-size10.size7{font-size:.57859209em}.katex .fontsize-ensurer.reset-size10.size8,.katex .sizing.reset-size10.size8{font-size:.69431051em}.katex .fontsize-ensurer.reset-size10.size9,.katex .sizing.reset-size10.size9{font-size:.83317261em}.katex .fontsize-ensurer.reset-size10.size10,.katex .sizing.reset-size10.size10{font-size:1em}.katex .fontsize-ensurer.reset-size10.size11,.katex .sizing.reset-size10.size11{font-size:1.19961427em}.katex .fontsize-ensurer.reset-size11.size1,.katex .sizing.reset-size11.size1{font-size:.20096463em}.katex .fontsize-ensurer.reset-size11.size2,.katex .sizing.reset-size11.size2{font-size:.24115756em}.katex .fontsize-ensurer.reset-size11.size3,.katex .sizing.reset-size11.size3{font-size:.28135048em}.katex .fontsize-ensurer.reset-size11.size4,.katex .sizing.reset-size11.size4{font-size:.32154341em}.katex .fontsize-ensurer.reset-size11.size5,.katex .sizing.reset-size11.size5{font-size:.36173633em}.katex .fontsize-ensurer.reset-size11.size6,.katex .sizing.reset-size11.size6{font-size:.40192926em}.katex .fontsize-ensurer.reset-size11.size7,.katex .sizing.reset-size11.size7{font-size:.48231511em}.katex .fontsize-ensurer.reset-size11.size8,.katex .sizing.reset-size11.size8{font-size:.57877814em}.katex .fontsize-ensurer.reset-size11.size9,.katex .sizing.reset-size11.size9{font-size:.69453376em}.katex .fontsize-ensurer.reset-size11.size10,.katex .sizing.reset-size11.size10{font-size:.83360129em}.katex .fontsize-ensurer.reset-size11.size11,.katex .sizing.reset-size11.size11{font-size:1em}.katex .delimsizing.size1{font-family:KaTeX_Size1}.katex .delimsizing.size2{font-family:KaTeX_Size2}.katex .delimsizing.size3{font-family:KaTeX_Size3}.katex .delimsizing.size4{font-family:KaTeX_Size4}.katex .delimsizing.mult .delim-size1>span{font-family:KaTeX_Size1}.katex .delimsizing.mult .delim-size4>span{font-family:KaTeX_Size4}.katex .nulldelimiter{display:inline-block;width:.12em}.katex .delimcenter,.katex .op-symbol{position:relative}.katex .op-symbol.small-op{font-family:KaTeX_Size1}.katex .op-symbol.large-op{font-family:KaTeX_Size2}.katex .accent>.vlist-t,.katex .op-limits>.vlist-t{text-align:center}.katex .accent .accent-body{position:relative}.katex .accent .accent-body:not(.accent-full){width:0}.katex .overlay{display:block}.katex .mtable .vertical-separator{display:inline-block;min-width:1px}.katex .mtable .arraycolsep{display:inline-block}.katex .mtable .col-align-c>.vlist-t{text-align:center}.katex .mtable .col-align-l>.vlist-t{text-align:left}.katex .mtable .col-align-r>.vlist-t{text-align:right}.katex .svg-align{text-align:left}.katex svg{fill:currentColor;stroke:currentColor;fill-rule:nonzero;fill-opacity:1;stroke-width:1;stroke-linecap:butt;stroke-linejoin:miter;stroke-miterlimit:4;stroke-dasharray:none;stroke-dashoffset:0;stroke-opacity:1;display:block;height:inherit;position:absolute;width:100%}.katex svg path{stroke:none}.katex img{border-style:none;max-height:none;max-width:none;min-height:0;min-width:0}.katex .stretchy{display:block;overflow:hidden;position:relative;width:100%}.katex .stretchy:after,.katex .stretchy:before{content:""}.katex .hide-tail{overflow:hidden;position:relative;width:100%}.katex .halfarrow-left{left:0;overflow:hidden;position:absolute;width:50.2%}.katex .halfarrow-right{overflow:hidden;position:absolute;right:0;width:50.2%}.katex .brace-left{left:0;overflow:hidden;position:absolute;width:25.1%}.katex .brace-center{left:25%;overflow:hidden;position:absolute;width:50%}.katex .brace-right{overflow:hidden;position:absolute;right:0;width:25.1%}.katex .x-arrow-pad{padding:0 .5em}.katex .cd-arrow-pad{padding:0 .55556em 0 .27778em}.katex .mover,.katex .munder,.katex .x-arrow{text-align:center}.katex .boxpad{padding:0 .3em}.katex .fbox,.katex .fcolorbox{border:.04em solid;box-sizing:border-box}.katex .cancel-pad{padding:0 .2em}.katex .cancel-lap{margin-left:-.2em;margin-right:-.2em}.katex .sout{border-bottom-style:solid;border-bottom-width:.08em}.katex .angl{border-right:.049em solid;border-top:.049em solid;box-sizing:border-box;margin-right:.03889em}.katex .anglpad{padding:0 .03889em}.katex .eqn-num:before{content:"(" counter(katexEqnNo) ")";counter-increment:katexEqnNo}.katex .mml-eqn-num:before{content:"(" counter(mmlEqnNo) ")";counter-increment:mmlEqnNo}.katex .mtr-glue{width:50%}.katex .cd-vert-arrow{display:inline-block;position:relative}.katex .cd-label-left{display:inline-block;position:absolute;right:calc(50% + .3em);text-align:left}.katex .cd-label-right{display:inline-block;left:calc(50% + .3em);position:absolute;text-align:right}.katex-display{display:block;margin:1em 0;text-align:center}.katex-display>.katex{display:block;text-align:center;white-space:nowrap}.katex-display>.katex>.katex-html{display:block;position:relative}.katex-display>.katex>.katex-html>.tag{position:absolute;right:0}.katex-display.leqno>.katex>.katex-html>.tag{left:0;right:auto}.katex-display.fleqn>.katex{padding-left:2em;text-align:left}body{counter-reset:katexEqnNo mmlEqnNo}

/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

.katex-error {
	color: var(--vscode-editorError-foreground);
}

</style>
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.2/dist/katex.min.css" integrity="sha384-yFRtMMDnQtDRO8rLpMIKrtPCD5jdktao2TV19YiZYWMDkUR5GQZR/NOVTdquEx1j" crossorigin="anonymous">
<link href="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.css" rel="stylesheet" type="text/css">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/markdown.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/highlight.css">
<style>
            body {
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe WPC', 'Segoe UI', system-ui, 'Ubuntu', 'Droid Sans', sans-serif;
                font-size: 14px;
                line-height: 1.6;
            }
        </style>
        <style>
.task-list-item { list-style-type: none; } .task-list-item-checkbox { margin-left: -20px; vertical-align: middle; }
</style>
        
    </head>
    <body class="vscode-body vscode-light">
        <h1 id="bio334-theory">BIO334 Theory</h1>
<p>David Parker + wikipedia</p>
<h1 id="table-of-contents">Table of Contents</h1>
<ul>
<li><a href="#bio334-theory">BIO334 Theory</a></li>
<li><a href="#table-of-contents">Table of Contents</a></li>
<li><a href="#mering">Mering</a>
<ul>
<li><a href="#multiple-sequence-alignment">Multiple sequence alignment</a>
<ul>
<li><a href="#motivations-for-sequence-alignment">Motivations for sequence alignment</a></li>
<li><a href="#how-its-done">How it’s done</a></li>
</ul>
</li>
<li><a href="#phylogenetic-trees">Phylogenetic trees</a>
<ul>
<li><a href="#distance-based-methods-phenetic-technique">Distance based methods (phenetic technique)</a></li>
<li><a href="#maximum-likelihood-cladistic-technique">Maximum Likelihood (cladistic technique)</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#shimizu">Shimizu</a>
<ul>
<li><a href="#genome-wide-polymorphisms">Genome-wide polymorphisms</a>
<ul>
<li><a href="#nucleotide-diversity">Nucleotide diversity</a></li>
<li><a href="#selective-sweep--hitchhiking">Selective sweep / hitchhiking</a></li>
<li><a href="#balancing-selection">Balancing selection</a></li>
<li><a href="#tajimas-d">Tajima’s D</a></li>
<li><a href="#positive-selection-on-replacement-mutations">positive selection on replacement mutations</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#wagner">Wagner</a>
<ul>
<li><a href="#metabolic-networks">Metabolic networks</a>
<ul>
<li><a href="#metabolic-genotype">Metabolic genotype</a></li>
<li><a href="#metabolic-phenotype">Metabolic phenotype</a></li>
<li><a href="#metabolic-flux">Metabolic flux</a></li>
<li><a href="#metabolic-models">Metabolic models</a></li>
<li><a href="#flux-balance-analysis">Flux balance analysis</a>
<ul>
<li><a href="#flux-balance-analysis-needs">Flux balance analysis needs</a></li>
<li><a href="#flux-balance-analysis-computes">Flux balance analysis computes</a></li>
<li><a href="#example-questions-for-flux-balance-analysis">Example questions for flux balance analysis</a></li>
</ul>
</li>
<li><a href="#stoichiometric-matrix">Stoichiometric matrix</a></li>
<li><a href="#essential-reactions-finding-reactions-that-must-be-there">Essential reactions: Finding reactions that must be there</a></li>
<li><a href="#active-reactions-finding-reactions-in-flux">Active reactions: Finding reactions in flux</a></li>
</ul>
</li>
<li><a href="#flux-variability-analysis-fva-finding-alternative-pathways">Flux Variability Analysis (FVA): Finding alternative pathways</a></li>
</ul>
</li>
<li><a href="#robinson">Robinson</a>
<ul>
<li><a href="#high-throughput-sequencing">High-throughput sequencing</a></li>
<li><a href="#single-cell">Single cell</a></li>
<li><a href="#cytometry">Cytometry</a>
<ul>
<li><a href="#flow-cytometry">Flow cytometry</a></li>
<li><a href="#mass-cytometry">Mass cytometry</a></li>
</ul>
</li>
<li><a href="#hierarchical-clustering">Hierarchical Clustering</a></li>
<li><a href="#dimension-reduction">Dimension reduction</a></li>
</ul>
</li>
</ul>
<h1 id="mering">Mering</h1>
<h2 id="multiple-sequence-alignment">Multiple sequence alignment</h2>
<h3 id="motivations-for-sequence-alignment">Motivations for sequence alignment</h3>
<ol>
<li>Find genes that are related by common descent</li>
<li>to identify and check the state of “active sites”</li>
<li>to identify and characterize “protein domains”</li>
<li>to make phylogenetic inferences (“trees”)</li>
</ol>
<h3 id="how-its-done">How it’s done</h3>
<ul>
<li>Substitution matrix
<ul>
<li>A substitution matrix describes the frequency at which a character in a nucleotide sequence or a protein sequence changes to other character states over evolutionary time.</li>
<li>Each amino acid is more or less likely to mutate into various other amino acids. For instance, a hydrophilic residue such as arginine is more likely to be replaced by another hydrophilic residue such as glutamine, than it is to be mutated into a hydrophobic residue such as leucine.</li>
</ul>
</li>
<li>Pairwise Alignment
<ul>
<li>BLAST: quick and dirty
<ul>
<li>basic local alignment search tool</li>
<li>Using a heuristic method, BLAST finds similar sequences, by locating short matches between the two sequences. This process of finding similar sequences is called seeding. It is after this first match that BLAST begins to make local alignments.</li>
</ul>
</li>
<li>Dynamic Programming: correct and slow
<ul>
<li>The Smith–Waterman algorithm performs local sequence alignment; that is, for determining similar regions between two strings of nucleic acid sequences or protein sequences. Instead of looking at the entire sequence, the Smith–Waterman algorithm compares segments of all possible lengths and optimizes the similarity measure.</li>
</ul>
</li>
</ul>
</li>
<li>Multiple Alignment
<ul>
<li>Combinatorial Explosion: very many possible solutions</li>
<li>Complexity: O(alignment_length^number of sequences)</li>
<li>=&gt; an NP-complete problem !
<ul>
<li>Although a solution to an NP-complete problem can be verified &quot;quickly&quot;, there is no known way to find a solution quickly. That is, the time required to solve the problem using any currently known algorithm increases rapidly as the size of the problem grows. As a consequence, determining whether it is possible to solve these problems quickly, called the P versus NP problem, is one of the fundamental unsolved problems in computer science today.</li>
<li>While a method for computing the solutions to NP-complete problems quickly remains undiscovered, computer scientists and programmers still frequently encounter NP-complete problems. NP-complete problems are often addressed by using heuristic methods and approximation algorithms.</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="phylogenetic-trees">Phylogenetic trees</h2>
<p>Generating phylogenetic trees</p>
<ul>
<li>Phenetic: trees are constructed based on observed characteristics directly, not on evolutionary history. (Distance based methods)</li>
<li>Cladistic: trees are constructed based on fitting observed characteristics to some model of evolutionary history (Maximum Likelihood methods)</li>
</ul>
<p>Which genes to use:</p>
<p>suitable marker genes ...</p>
<ul>
<li>should occur in every organism</li>
<li>should rarely undergo horizontal transfer</li>
<li>should be evolving ‘slowly’</li>
<li>should only occur in one copy per genome</li>
<li>should function in a process that sees no change</li>
</ul>
<p>For old events:</p>
<ul>
<li>Ribosomes and polymerases.</li>
</ul>
<p>For recent events:</p>
<ul>
<li>Fast evolving genes.</li>
</ul>
<h3 id="distance-based-methods-phenetic-technique">Distance based methods (phenetic technique)</h3>
<p>UPGMA: unweighted pair group method with arithmetic mean</p>
<ol>
<li>Alignment</li>
<li>Distance matrix</li>
<li>Choose the smallest distance</li>
<li>Join them</li>
<li>Calculate the new distances to every other node in the tree
<ol>
<li>We do this with a simple average of distances</li>
<li>Dist[Spinach, MonHum]
<ol>
<li>= (Dist[Spinach, Monkey] + Dist[Spinach, Human])/2</li>
<li>= (90.8 + 86.3)/2 = 88.55</li>
</ol>
</li>
</ol>
</li>
<li>Repeat until all nodes are joined</li>
</ol>
<h3 id="maximum-likelihood-cladistic-technique">Maximum Likelihood (cladistic technique)</h3>
<ul>
<li>The likelihood is the probability of the data given the model</li>
<li>The probability of observing the data under the assumed model will change depending on the parameter values of the model.</li>
<li>The aim of maximum likelihood is to choose the value of the parameter that maximizes the probability of finding the data.</li>
</ul>
<p>Typically, the model has additional free ‘parameters’:</p>
<ul>
<li>The rate of evolution can vary across parts of the tree</li>
<li>The rate of evolution can vary from site to site in the protein</li>
</ul>
<p>maximum likelihood is computed like so:</p>
<ol>
<li>Image all ancestral possibilities and evolutionary paths.</li>
<li>Compute the likelihood of each path
<ul>
<li>L(path) = L(root) x <span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi mathvariant="normal">Π</mi></mrow><annotation encoding="application/x-tex">\Pi</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.6833em;"></span><span class="mord">Π</span></span></span></span> L(branches)</li>
<li>=P(G→T)P(G→G) P(G→A)P(G→G) [...]</li>
</ul>
</li>
<li>Multiply all likelihoods over all possible paths</li>
<li>Throughout, do not forget to optimize all free parameters</li>
<li>Repeat for each tree topology, identify the one with best Likelihood</li>
</ol>
<p>How do we verify a tree?</p>
<ul>
<li>Simulation</li>
<li>Bootstrapping
<ul>
<li>Bootstrap involves resampling with replacement from one’s molecular data with to create fictional datasets, called bootstrap replicates, of the same size. Specifically, the molecular data is typically organized as a multiple sequence alignment (MSA) of s species ×n characters. Since most models assume independent characters, we generate a replicate by sampling n characters, with replacement, from the original MSA and do this B times. <a href="https://doi.org/10.1016/B978-0-323-89775-4.00025-0">Krishna Kumar Ojha et al. (2022)</a></li>
</ul>
</li>
</ul>
<h1 id="shimizu">Shimizu</h1>
<h2 id="genome-wide-polymorphisms">Genome-wide polymorphisms</h2>
<p>In genomics, a genome-wide association study (GWA study, or GWAS), also known as whole genome association study (WGA study, or WGAS), is an observational study of a genome-wide set of genetic variants in different individuals to see if any variant is associated with a trait. GWA studies typically focus on associations between single-nucleotide polymorphisms (SNPs) and traits like major human diseases, but can equally be applied to any other genetic variants and any other organisms.</p>
<h3 id="nucleotide-diversity">Nucleotide diversity</h3>
<p>Nucleotide diversity is a measure of genetic variation. It is usually associated with other statistical measures of population diversity, and is similar to expected heterozygosity. This statistic may be used to monitor diversity within or between ecological populations, to examine the genetic variation in crops and related species, or to determine evolutionary relationships.</p>
<h3 id="selective-sweep--hitchhiking">Selective sweep / hitchhiking</h3>
<p>In genetics, a selective sweep is the process through which a new beneficial mutation that increases its frequency and becomes fixed (i.e., reaches a frequency of 1) in the population leads to the reduction or elimination of genetic variation among nucleotide sequences that are near the mutation. In selective sweep, positive selection causes the new mutation to reach fixation so quickly that linked alleles can &quot;hitchhike&quot; and also become fixed.</p>
<h3 id="balancing-selection">Balancing selection</h3>
<p>Balancing selection refers to a number of selective processes by which multiple alleles are actively maintained in the gene pool of a population at frequencies larger than expected from genetic drift alone. This can happen by various mechanisms, in particular, when the heterozygotes for the alleles under consideration have a higher fitness than the homozygote. In this way genetic polymorphism is conserved.</p>
<h3 id="tajimas-d">Tajima’s D</h3>
<p>Tajima's D is computed as the difference between two measures of genetic diversity: the mean number of pairwise differences and the number of segregating sites, each scaled so that they are expected to be the same in a neutrally evolving population of constant size.</p>
<p>The purpose of Tajima's D test is to distinguish between a DNA sequence evolving randomly (&quot;neutrally&quot;) and one evolving under a non-random process, including directional selection or balancing selection, demographic expansion or contraction, genetic hitchhiking, or introgression. A randomly evolving DNA sequence contains mutations with no effect on the fitness and survival of an organism. The randomly evolving mutations are called &quot;neutral&quot;, while mutations under selection are &quot;non-neutral&quot;. For example, a mutation that causes prenatal death or severe disease would be expected to be under selection. In the population as a whole, the frequency of a neutral mutation fluctuates randomly (i.e. the percentage of individuals in the population with the mutation changes from one generation to the next, and this percentage is equally likely to go up or down) through genetic drift.</p>
<p><strong>Tajimas D &gt; 0</strong></p>
<p><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>π</mi><mo>&gt;</mo><mi>s</mi><mi mathvariant="normal">/</mi><msubsup><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>n</mi><mo>−</mo><mn>1</mn></mrow></msubsup><mfrac><mn>1</mn><mi>i</mi></mfrac></mrow><annotation encoding="application/x-tex">\pi &gt; s/\sum_{i=1}^{n-1}{\frac{1}{i}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.5782em;vertical-align:-0.0391em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">π</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">&gt;</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.299em;vertical-align:-0.345em;"></span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mop"><span class="mop op-symbol small-op" style="position:relative;top:0em;">∑</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.954em;"><span style="top:-2.4003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mrel mtight">=</span><span class="mord mtight">1</span></span></span></span><span style="top:-3.2029em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">n</span><span class="mbin mtight">−</span><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2997em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord"><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.8451em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.394em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.345em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></p>
<ul>
<li>Many higher heterozygous sites
<ul>
<li>Population size decreases rapidly (bottleneck effect, founder effect)
<ul>
<li>singleton may be removed from the population</li>
</ul>
</li>
<li>Balancing selection (heterozygote advantage)</li>
</ul>
</li>
</ul>
<p><strong>Tajimas D &lt; 0</strong></p>
<p><span class="katex"><span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>π</mi><mo>&lt;</mo><mi>s</mi><mi mathvariant="normal">/</mi><msubsup><mo>∑</mo><mrow><mi>i</mi><mo>=</mo><mn>1</mn></mrow><mrow><mi>n</mi><mo>−</mo><mn>1</mn></mrow></msubsup><mfrac><mn>1</mn><mi>i</mi></mfrac></mrow><annotation encoding="application/x-tex">\pi &lt; s/\sum_{i=1}^{n-1}{\frac{1}{i}}</annotation></semantics></math></span><span class="katex-html" aria-hidden="true"><span class="base"><span class="strut" style="height:0.5782em;vertical-align:-0.0391em;"></span><span class="mord mathnormal" style="margin-right:0.03588em;">π</span><span class="mspace" style="margin-right:0.2778em;"></span><span class="mrel">&lt;</span><span class="mspace" style="margin-right:0.2778em;"></span></span><span class="base"><span class="strut" style="height:1.299em;vertical-align:-0.345em;"></span><span class="mord mathnormal">s</span><span class="mord">/</span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mop"><span class="mop op-symbol small-op" style="position:relative;top:0em;">∑</span><span class="msupsub"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.954em;"><span style="top:-2.4003em;margin-left:0em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span><span class="mrel mtight">=</span><span class="mord mtight">1</span></span></span></span><span style="top:-3.2029em;margin-right:0.05em;"><span class="pstrut" style="height:2.7em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">n</span><span class="mbin mtight">−</span><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.2997em;"><span></span></span></span></span></span></span><span class="mspace" style="margin-right:0.1667em;"></span><span class="mord"><span class="mord"><span class="mopen nulldelimiter"></span><span class="mfrac"><span class="vlist-t vlist-t2"><span class="vlist-r"><span class="vlist" style="height:0.8451em;"><span style="top:-2.655em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mathnormal mtight">i</span></span></span></span><span style="top:-3.23em;"><span class="pstrut" style="height:3em;"></span><span class="frac-line" style="border-bottom-width:0.04em;"></span></span><span style="top:-3.394em;"><span class="pstrut" style="height:3em;"></span><span class="sizing reset-size6 size3 mtight"><span class="mord mtight"><span class="mord mtight">1</span></span></span></span></span><span class="vlist-s">​</span></span><span class="vlist-r"><span class="vlist" style="height:0.345em;"><span></span></span></span></span></span><span class="mclose nulldelimiter"></span></span></span></span></span></span></p>
<ul>
<li>Many lower heterozygous sites
<ul>
<li>Population size increases rapidly
<ul>
<li>Singleton may be introduced in the population</li>
</ul>
</li>
<li>Positive/Negative selection (purifying selection, selective sweep, directional selection)</li>
</ul>
</li>
</ul>
<h3 id="positive-selection-on-replacement-mutations">positive selection on replacement mutations</h3>
<p>??</p>
<!-- ## Calculations

Nucleotide diversity2 [bio334_day2_part1.ipynb](jupyter_notebooks/bio334_day2_part1.ipynb) [bio334_day2_part2.ipynb](jupyter_notebooks/bio334_day2_part2.ipynb)

Tajima's D calculation1 [bio334_day2_part3.ipynb](jupyter_notebooks/bio334_day2_part3.ipynb) [bio334_day2_part4.ipynb](jupyter_notebooks/bio334_day2_part4.ipynb)

Tajima's D calculation2 [bio334_day2_part5.ipynb](jupyter_notebooks/bio334_day2_part5.ipynb) -->
<h1 id="wagner">Wagner</h1>
<h2 id="metabolic-networks">Metabolic networks</h2>
<p>A metabolic network is a network of chemical reactions whose two main functions are to produce</p>
<ul>
<li>chemical energy (for maintenance of cell functions and for biosynthesis)</li>
<li>molecular building blocks for biosynthesis</li>
</ul>
<h3 id="metabolic-genotype">Metabolic genotype</h3>
<p>The part of a genome that encodes metabolic enzymes</p>
<h3 id="metabolic-phenotype">Metabolic phenotype</h3>
<p><strong>Most general</strong>: All the molecules that a metabolism can synthesize
in a given chemical environment</p>
<p>The most important of these molecules are biomass precursors
(amino acids, DNA and RNA building blocks etc.).</p>
<p>A metabolism is viable if it can synthesize all of them.</p>
<p><strong>More specific:</strong> the spectrum of nutrients on which a metabolism is
viable</p>
<h3 id="metabolic-flux">Metabolic flux</h3>
<p>The rate at which an enzyme converts substrate into product per unit time.</p>
<h3 id="metabolic-models">Metabolic models</h3>
<p>Metabolic models include artificial reactions that allow the flux of metabolites in and out
of the system: external aka exchange reactions. They act on external metabolites, that is,
a metabolite that is outside a cell that can be transported in or out of a cell. By
convention, the role of an external reaction is to remove metabolites from the metabolic
network, so an influx of a given metabolite into the network carries a negative flux. You
may have already noticed that external reactions are necessary for FBA to work –
metabolites have to be introduced somewhere and taken out of the system somewhere else
if there is to be flux with unchanging metabolite concentrations.</p>
<p>For that very reason, external reactions are a convenient way to specify the kind of
environment in which we want to simulate cell growth. By changing the lower bounds of
an exchange reaction (remember influx carries a negative flux!) you can define what
metabolites are available to a cell.  Note that external reactions are different from
transport reactions as the latter allow import/export of metabolites into the cell. As you
get to more complex metabolic networks, you will see that cells often have multiple ways
of taking up or excreting metabolites. For example, in the toy model (Figure 1) you may
have noticed that there are two pathways for the uptake of metabolite B_e. If you wanted
to change how much of B_e the cell takes up, you don’t need to change the bounds of
both transport reactions R7 and R16; you can simply change the bounds of the external
reaction and leave it to FBA to decide which pathway is used for the uptake of B_e.</p>
<h3 id="flux-balance-analysis">Flux balance analysis</h3>
<p>FBA is a method to predict the flux of material through a metabolic network that
maximizes the flux through a target reaction given two kinds of constraints:</p>
<ul>
<li>The first one arises from the relative proportions of reactants and
products in chemical reactions (given in the stoichiometric matrix)</li>
<li>the other is how much flux a reaction is allowed to carry (reaction bounds).</li>
</ul>
<p>FBA is a very powerful tool in the analysis of metabolic networks, and also computationally efficient even for
genome-scale models, as it assumes the network is in steady state – metabolites do not
accumulate, they are produced as fast as they are consumed and consumed as fast as they
are produced.</p>
<h4 id="flux-balance-analysis-needs">Flux balance analysis needs</h4>
<ol>
<li>a list of chemical reactions known to be catalyzed by enzymes in a given organism</li>
<li>Information about nutrients in the chemical environment of a cell and their uptake rate (usually in mol/g dry weight [DW] and hour)</li>
</ol>
<h4 id="flux-balance-analysis-computes">Flux balance analysis computes</h4>
<ol>
<li>allowable metabolic fluxes through a metabolic network (fluxes that do not violate the law of mass conservation)</li>
<li>within the set of allowable fluxes, those that have desirable properties (e.g., maximal rate of biomass production, maximal biomass yield per unit of carbon source.</li>
</ol>
<h4 id="example-questions-for-flux-balance-analysis">Example questions for flux balance analysis</h4>
<ol>
<li>Can a given organism (metabolism) survive in environment X?</li>
<li>How fast could it grow in this environment?</li>
<li>Why are many enzymatic reactions dispensable in any one environment?</li>
<li>Why do some metabolisms have many reactions, while others have few?</li>
<li>Does network function and flux influence network evolution?</li>
<li>Is it possible to design “resistance-proof” antimetabolic drugs?</li>
</ol>
<h3 id="stoichiometric-matrix">Stoichiometric matrix</h3>
<p>Stoichiometric coefficients represent the relative molar amounts of reactants (educts,
substrates) and products participating in a chemical reaction. For example, in the
reaction A + 2 B -&gt; C, the stoichiometric coefficient of A is 1, that of B is 2 and that of C
is 1. To denote whether a molecule is a reactant or a product of a reaction, we add a sign
to its stoichiometric coefficient. Thus, the stoichiometry of A is -1, the stoichiometry of B
is -2 and that of C is 1.</p>
<h3 id="essential-reactions-finding-reactions-that-must-be-there">Essential reactions: Finding reactions that must be there</h3>
<p>A reaction is essential for the synthesis of a molecule from a specific substrate if its
removal makes the synthesis of that molecule from the substrate impossible. As an
example, consider the reactions involved in the synthesis of metabolite O from substrate
B_e in figure 1. Synthesis of O from B_e requires reactions R8 and R9, but not R7 and
R15, as these can be replaced by R16. If we were to delete either R8 or R9, it would be
impossible to synthesize O from B_e. We therefore say that reactions R8 and R9 are
essential for the production of O from substrate B_e.</p>
<h3 id="active-reactions-finding-reactions-in-flux">Active reactions: Finding reactions in flux</h3>
<p>Active reactions are reactions that have a non-zero metabolic flux, that is, reactions
proceeding at a rate different from zero. It is NOT the upper and/or lower bound of a
reaction that determines whether it is active, but the effective flux of the reaction in the
FBA solution. Therefore, you can only know whether a reaction is active or not after
performing FBA.</p>
<h2 id="flux-variability-analysis-fva-finding-alternative-pathways">Flux Variability Analysis (FVA): Finding alternative pathways</h2>
<p>FBA solutions are rarely unique. For example, think about the active pathway you found
in exercise 1.6 that produces X_e from A_e. Is this the only possible solution that
maximizes production of X_e? Pay particular attention to reactions R3 and R4.<br>
In general, we find that at least some reactions can take a whole range of fluxes without
affecting the flux through the objective function. Metabolic networks are thus “flux
variable”, and FVA can provide some insights into the extent of the network’s flux
variability. FVA computes the minimum and maximum value of the flux through a
reaction while keeping a given objective, such as biomass synthesis, unchanged, thus
estimating the range of possible fluxes through a reaction (the reaction’s flux variability).</p>
<h1 id="robinson">Robinson</h1>
<p>It's just data</p>
<h2 id="high-throughput-sequencing">High-throughput sequencing</h2>
<p>Gene Expression Profiling: questions of interest</p>
<ul>
<li>What genes have changed in expression? (e.g. between disease/normal, affected by treatment)
<ul>
<li>Gene discovery, differential expression</li>
</ul>
</li>
<li>Is a specified group of genes all up-regulated in a particular condition?
<ul>
<li>Gene set differential expression</li>
</ul>
</li>
<li>Can the expression profile predict outcome?
<ul>
<li>Class prediction, classification</li>
</ul>
</li>
<li>Are there tumour sub-types not previously identified? Do my genes group into previously undiscovered pathways?
<ul>
<li>Class discovery, clustering</li>
</ul>
</li>
<li>(with single cell gene expression): What changes in cell type composition are observed? What genes have changed in expression in a given subtype of cells?</li>
</ul>
<h2 id="single-cell">Single cell</h2>
<p>Bulk vs Single Cell RNA Sequencing (scRNA-seq):</p>
<ul>
<li>
<p>Bulk RNA-seq: average expression level</p>
<ul>
<li>comparative transcriptomics</li>
<li>disease biomarker</li>
<li>homogenous systems</li>
</ul>
</li>
<li>
<p>scRNA-seq: expression level of each single cell</p>
<ul>
<li>define heterogeneity</li>
<li>identify rare cell population</li>
<li>cell population dynamics</li>
<li>finding cell subpopulation-specific changes in state</li>
</ul>
</li>
</ul>
<h2 id="cytometry">Cytometry</h2>
<h3 id="flow-cytometry">Flow cytometry</h3>
<p>In this process, a sample containing cells or particles is suspended in a fluid and injected into the flow cytometer instrument. The sample is focused to ideally flow one cell at a time through a laser beam, where the light scattered is characteristic to the cells and their components. Cells are often labeled with fluorescent markers so light is absorbed and then emitted in a band of wavelengths. Tens of thousands of cells can be quickly examined and the data gathered are processed by a computer.</p>
<ul>
<li>Cell counting</li>
<li>Cell sorting</li>
<li>Determining cell characteristics and function</li>
<li>Detecting microorganisms</li>
<li>Biomarker detection</li>
<li>Protein engineering detection</li>
<li>Diagnosis of health disorders such as blood cancers</li>
</ul>
<h3 id="mass-cytometry">Mass cytometry</h3>
<p>Mass cytometry is a mass spectrometry technique based on inductively coupled plasma mass spectrometry and time of flight mass spectrometry used for the determination of the properties of cells (cytometry). In this approach, antibodies are conjugated with isotopically pure elements, and these antibodies are used to label cellular proteins. Cells are nebulized and sent through an argon plasma, which ionizes the metal-conjugated antibodies. The metal signals are then analyzed by a time-of-flight mass spectrometer. The approach overcomes limitations of spectral overlap in flow cytometry by utilizing discrete isotopes as a reporter system instead of traditional fluorophores which have broad emission spectra.</p>
<ul>
<li>Finding molecular biomarkers associated with drug response</li>
<li>Differential abundance of cell populations</li>
</ul>
<h2 id="hierarchical-clustering">Hierarchical Clustering</h2>
<ul>
<li>Divisive: This is a &quot;top-down&quot; approach: all observations start in one cluster, and splits are performed recursively as one moves down the hierarchy.</li>
<li>Agglomerative: This is a &quot;bottom-up&quot; approach: each observation starts in its own cluster, and pairs of clusters are merged as one moves up the hierarchy.</li>
</ul>
<p><strong>Metric</strong>: The choice of an appropriate metric will influence the shape of the clusters, as some elements may be relatively closer to one another under one metric than another. For example, in two dimensions, under the Manhattan distance metric, the distance between the origin (0,0) and (0.5, 0.5) is the same as the distance between the origin and (0, 1), while under the Euclidean distance metric the latter is strictly greater.</p>
<p><strong>Linkage</strong>: The linkage criterion determines the distance between sets of observations as a function of the pairwise distances between observations.</p>
<h2 id="dimension-reduction">Dimension reduction</h2>
<ul>
<li>Many types of data come as a matrix of N samples (e.g., cells, patients) x G features (e.g., genes, proteins)</li>
<li>Each sample is a point in G-dimensional space</li>
<li>Goal: represent the data in 2-3 dimensions, but preserve structure as best as possible (i.e., points that are close in G dimensions should be close in 2 or 3 dimensions)</li>
</ul>
<p>PCA: Principal Component Analysis (PCA) is a statistical method for dimensionality reduction.</p>
<ul>
<li>Form successive linear combinations of the features that are: orthogonal, ordered by variance</li>
</ul>

        <script async src="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.js"></script>
        
    </body>
    </html>